### PR TITLE
vote button implemented

### PR DESCRIPTION
**Pull request by peratik**
## Changes
Vote button implemented

## Checklist

- [x] I've performed a self-review, followed all rules from [Frontend style guide](https://www.notion.so/santiment/Front-end-style-guide-81750096b38c4bea9a29b14fd4ab8667)
- [x] If I make changes to another person's module, I've asked how to use it or request a review
- [x] I've updated the [documentation](https://github.com/santiment/academy), if necessary (Keyboard shortcuts, Account settings)
- [x] I've checked night mode, mobile & tablet screens (if have changes in UI)
- [x] I've added tests (if necessary)

## Screenshots or GIFs

image

### CODE REVIEW

File: src/pages/Explorer/Category/UserCreation.svelte [hunk 1]
```diff
@@ -21,6 +24,10 @@
         class="$style.actions"
         isOwner={$currentUser && user && user.id === $currentUser.id}
         {url}
+        {item}
+        {type}
+        showCommentAction={commentsCount >= 0}
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Category/UserCreation.svelte:29]: `commentsCount` can't be negative, hence this expression always results in `true`
```suggestion
        showCommentAction
```

File: src/pages/Explorer/Category/UserCreation.svelte [hunk 2]
```diff
@@ -6,12 +6,15 @@
...
 
-  export let item
+  export let item = {}
   export let small = false
   export let showActions = false
   export let url
+  export let type
 
-  $: ({ user, votes, commentsCount } = item)
+  let totalVotes = item.votes.totalVotes
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Category/UserCreation.svelte:15]: Should be reactive statement.

File: src/pages/Explorer/Components/Actions.svelte [hunk 3]
```diff
@@ -1,20 +1,51 @@
...
 
   let className = ''
   export { className as class }
 
   export let isOwner = false
   export let isModerator = false
   export let url
+  export let item = {}
+  export let type
+  export let onVoteCountChange = console.log
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/Actions.svelte:15]: Remove `console.log`

File: src/pages/Explorer/Components/Actions.svelte [hunk 4]
```diff
@@ -1,20 +1,51 @@
...
+      mutation {
+        vote(${voteType}:${id}) { 
+          votes {
+            totalVotes
+          }
+        }
+      }
+    `
+
+    mutate(VOTE_MUTATION).then(({ vote: { votes } }) => {
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/Actions.svelte:40]: Move to `/api/vote.js`

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/pages/Explorer/Category/UserCreation.svelte (hunk 1 above)
Comment (peratik) [reply src/pages/Explorer/Category/UserCreation.svelte:29]: It can be null

Comment (peratik) [reply src/pages/Explorer/Category/UserCreation.svelte:29]: alerts don't have comments and that field is null

Comment (DmitriVanGuard) [reply src/pages/Explorer/Category/UserCreation.svelte:28]: Makes sense 👍

**Review by DmitriVanGuard — APPROVED** [review]

**Pull request merged**